src/Api/EncodedStringTest.cpp

```diff
@@ -0,0 +1,521 @@
+  decoded_string.push_back(0x66);
+  decoded_string.push_back(0x77);
+  decoded_string.push_back(0x88);
+
+  decoded_string.push_back(0x11);
+  decoded_string.push_back(0x22);
+  decoded_string.push_back(0x33);
+  decoded_string.push_back(0x44);
+  decoded_string.push_back(0x55);
+  decoded_string.push_back(0x66);
+  decoded_string.push_back(0x77);
+  decoded_string.push_back(0x88);
+
+  decoded_string.push_back(0x11);
+  decoded_string.push_back(0x22);
+  decoded_string.push_back(0x33);
+  decoded_string.push_back(0x44);
+  decoded_string.push_back(0x55);
+  decoded_string.push_back(0x66);
+  decoded_string.push_back(0x77);
+  decoded_string.push_back(0x88);
+
+  decoded_string.push_back(0x11);
+  decoded_string.push_back(0x22);
+  decoded_string.push_back(0x33);
+  decoded_string.push_back(0x44);
+  decoded_string.push_back(0x55);
+  decoded_string.push_back(0x66);
+  decoded_string.push_back(0x77);
+  decoded_string.push_back(0x88);
+
+  decoded_string.push_back(0x88);
+  decoded_string.push_back(0x77);
+  decoded_string.push_back(0x66);
+  decoded_string.push_back(0x55);
+  decoded_string.push_back(0x44);
+  decoded_string.push_back(0x33);
+  decoded_string.push_back(0x22);
+  decoded_string.push_back(0x11);
+
+  EncodeString(decoded_string.c_str(), &encoded_string);
+  EXPECT_EQ(encoded_string.encoded_name_1(), 0x8877665544332211);
+  EXPECT_EQ(encoded_string.encoded_name_2(), 0x8877665544332211);
+  EXPECT_EQ(encoded_string.encoded_name_3(), 0x8877665544332211);
+  EXPECT_EQ(encoded_string.encoded_name_4(), 0x8877665544332211);
+  EXPECT_EQ(encoded_string.encoded_name_5(), 0x8877665544332211);
+  EXPECT_EQ(encoded_string.encoded_name_6(), 0x8877665544332211);
+  EXPECT_EQ(encoded_string.encoded_name_7(), 0x8877665544332211);
+  EXPECT_EQ(encoded_string.encoded_name_8(), 0x8877665544332211);
+  EXPECT_THAT(encoded_string.encoded_name_additional(),
+              ElementsAre(0x8877665544332211, 0x1122334455667788));
+}
+
+TEST(EncodedString, SmokeSmallString) {
+  std::string expected_string = "Some short string";
+  ApiScopeStart encoded_string;
+  EncodeString(expected_string.c_str(), &encoded_string);
+  std::string decoded_string =
+      DecodeString(encoded_string.encoded_name_1(), encoded_string.encoded_name_2(),
+                   encoded_string.encoded_name_3(), encoded_string.encoded_name_4(),
+                   encoded_string.encoded_name_5(), encoded_string.encoded_name_6(),
+                   encoded_string.encoded_name_7(), encoded_string.encoded_name_8(),
+                   encoded_string.encoded_name_additional().data(),
+                   encoded_string.encoded_name_additional_size());
+
+  EXPECT_EQ(expected_string, decoded_string);
+}
+
+TEST(EncodedString, SmokeLargeString) {
+  std::string expected_string =
+      "na na na na na na na na na na na na na na na na na na na na na na na na na na na na na na "
+      "na na na na na na na na na nana na na na na na na na na na na na na na na na na na na na "
```

Comment:
The "nana" without the space makes me uncomfortable :D

Reply:
Done :)